### ISSUE TITLE
Cloudless vs Nextcloud on F-Droid

### ISSUE THREAD

**Issue by smichel17** [mpcjanssen/simpletask-android#816]
I previously had ST Cloudless installed through F-Droid. In F-Droid, it was called simply "Simpletask". The launcher was called "Simpletask Cloudless".

As of the most recent version or two, it now appears as "Simpletask Nextcloud" in F-Droid and "ST Cloudless" in the launcher. The actual installed app remains cloudless, though I have not tried doing a fresh install.

**Comment by Bubu** [comment]
I think it's related to f-droid now pulling the apop metadata from this repo's directory structure according to this layout: https://github.com/Triple-T/gradle-play-publisher#play-store-metadata

apparently it falls over the multiple flavours.

**Comment by schiessle** [comment]
Is there any reason why the Nextcloud version is not build for F-Droid? I would love to use it.

**Comment by mpcjanssen** [comment]
@schiessle not any particular reason except that it requires work. Considering you can relatively easy build yourself this is not a priority for me now.

**Comment by feluxe** [comment]
This is really confusing... I installed "Simpletask Nextcloud" via F-Droid. I thought this had to work with Nextcloud (since it's in the name). It took me an hour before I landed here and realized that I had installed the wrong app.

**Comment by mpcjanssen** [comment]
@feluxe I agree it's confusing. Unfortunately there is not much I can do about it.

**Comment by mpcjanssen** [comment]
Seems fixed in current F-Droid version

**Comment by kirschner** [comment]
@mpcjanssen to clarify: the version which is now in F-Droid does not support Nextcloud but is the standalone version, right?

**Comment by mpcjanssen** [comment]
Correct

**Comment by kirschner** [comment]
Thanks for clarifications @mpcjanssen I would still be interested in a version with nextcloud support in f-droid, but I can understand that it is not a priority for you. Thanks for your work!

**Comment by mpcjanssen** [comment]
Feel free to add it to F-Droid yourself. The only reason for me not to added is lack of time. If someone will make the required fdroiddata PR I am all for it.

**Comment by fazlerabbi37** [comment]
> If someone will make the required fdroiddata PR I am all for it.

hi @mpcjanssen, I know this is a very old issue but I still am interested in getting the Nextcloud variant in F-Droid. I may be able to help with the _fdroiddata PR, but would require a lot of your help. I have a couple of queries first:

- It is unclear to me how are you building the Nextcloud app from the same source and what is added in that build.
- What about the package name? Do you want to keep it as same as the Play Store one? ie: `nl.mpcjanssen.simpletask.nextcloud`

**Comment by mpcjanssen** [comment]
@fazlerabbi37 there is already a PR for the webdav version which can also be used for nextcloud see https://gitlab.com/fdroid/fdroiddata/-/merge_requests/6534

**Comment by fazlerabbi37** [comment]
@mpcjanssen I just copied and pasted your MR, changed the app name to Simpletask Nextcloud and created a new MR: https://gitlab.com/fdroid/fdroiddata/-/merge_requests/6680. Please give it a look and let me know if any improvement is needed.

**Comment by smichel17** [comment]
Looks like it should work. Note that there are instructions in the fdroiddata repo for verifying that it actually will work (by downloading the fdroid server and trying to run it locally; it's not that hard).

**Comment by fazlerabbi37** [comment]
@mpcjanssen I just hit a bit of problem regarding the [RFP](https://gitlab.com/fdroid/fdroiddata/-/merge_requests/6680#note_352363278) for Nextcloud flavor. Simpletask usages [Triple-T structure](https://github.com/mpcjanssen/simpletask-android/tree/master/app/src/main/play) to provide the description and summery but for the Cloudless app in F-Droid almost all [metadata](https://gitlab.com/fdroid/fdroiddata/-/tree/master/metadata/nl.mpcjanssen.simpletask) are acompanied in the `fdroiddata` repo. 

My question what should I do for the Nextcloud flavor? I can create a PR to update the Triple-T structure with the data from `fdroiddata` repo and remove the data from there for Cloudless flavor or just copy and change the `fdroiddata` repo data for the Nextcloud flavor. 

The F-Droid maintainers are pushing (gently) towards adopting either Fastlane or Triple-T file structure as it is less hassle for them and much more easy to maintain for the app developer. IMO, we should go with the first option as you already have Triple-T. Please keep in mind that if we do so, you need to tag and make a new release for it to be picked up by F-Droid.

**Comment by mpcjanssen** [comment]
If you could create PR for Triple-T that would be great.

**Comment by smichel17** [comment]
[Meta] it's annoying to me that neither fastlane nor triple-T get  everything right.

Triple-T:

- Metadata inside app folders makes sense
- Gradle plugin instead of standalone tool is great
- Metadata structure leaves much to be desired.

Fastlane:


- Metadata structure works well with multiple locales
- Changelog structure makes way more sense

Fastlane structure in Triple-T location as a gradle plugin would be amazing.

**Comment by smichel17** [comment]
Follow-up: https://gitlab.com/fdroid/fdroidserver/-/issues/829

**Comment by fazlerabbi37** [comment]
@mpcjanssen I just hard back from F-Droid about this issue. As per this [comment](https://gitlab.com/fdroid/fdroiddata/-/merge_requests/6680#note_474032844), the way fastlane is structured now will not work with F-Droid. In order to get it working, it needs to be flavor-specific. Do you want me to restructure the fastlane directory and send a new PR?

**Comment by mpcjanssen** [comment]
@fazlerabbi37 if you could restructure it, that would be great.

**Comment by fazlerabbi37** [comment]
I will send a PR ASAP.

**Comment by smichel17** [comment]
Aaah, I also need to get around to the f-droid metadata improvement, it's been on my list for foreeever :S

**Comment by fazlerabbi37** [comment]
I just created the PR to move Fastlane file to flavor-specific locations #1115.
---
Now I have a question for @mpcjanssen and @IzzySoft, what happens to the `fastlane` directory and the files inside them? I am assuming they will stay there, but can we remove the `metadata` directory inside `fastlane` directory? I have kept them for now but can update the PR after removing them, if it is ok to do so.

**Comment by IzzySoft** [comment]
Just be clear of what you're asking: No need to keep the stuff inside `/fastlane` (i.e. the previous/old location) for F-Droid, we don't need duplicates; the one in the new location of course needs the `metadata` (i.e. `app/src/*/fastlane/metadata` must stay :wink:). When the PR is merged, we should send the bot again on F-Droid's end to see it finally finds it (which might need a tag "on top").

**Comment by mpcjanssen** [comment]
@fazlerabbi37 PR merged, thanks.

**Comment by fazlerabbi37** [comment]
update on F-Droid side: as per this [comment](https://gitlab.com/fdroid/fdroiddata/-/merge_requests/6680#note_547158125) the need a new release. @mpcjanssen can you please take a look?

**Comment by mpcjanssen** [comment]
@fazlerabbi37 I have tagged 10.9.1.

**Comment by fazlerabbi37** [comment]
update on F-Droid side: RFP PR is merged! :tada: Hopefully will be build within next week. Thank you, @mpcjanssen, @IzzySoft and everyone!

PS: @mpcjanssen may need additional help from you to do some additional fix and cleanup after the app is published.

**Comment by mpcjanssen** [comment]
@fazlerabbi37 thanks a lot. Please let me know if you need something else (log a new issue)
